### PR TITLE
Book: Fix phrasing: “an associated type” → “an object with an associated type”.

### PR DESCRIPTION
**Pull request by progval**
From what I understood, `graph` is the object from which we create a trait object, and the associated types are `Graph::N` and `Graph::E`.

### CODE REVIEW

File: src/doc/book/associated-types.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ declarations.
 ## Trait objects with associated types
 
 There’s one more bit of syntax we should talk about: trait objects. If you
-try to create a trait object from an associated type, like this:
+try to create a trait object from an object with an associated type, like this:
```
Comment (apasel422) [inline src/doc/book/associated-types.md]: I would say that "from a trait with an associated type" is somewhat more accurate.

Comment (steveklabnik) [inline src/doc/book/associated-types.md]: Agreed

Comment (ProgVal) [inline src/doc/book/associated-types.md]: Changed.

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thank you!

**Pull request merged**